specs/schemas/schema-schema.ipldsch

```diff
@@ -336,14 +341,30 @@ type TypeLink struct {
 ## Unions can be defined as representing in several different ways: see
 ## the documentation on the UnionRepresentation type for details.
 ##
-## The set of types which the union can contain are specified in a map
-## inside the representation field.  (The key type of the map varies per
-## representation strategy, so it's not possible to keep on this type directly.)
-##
 type TypeUnion struct {
+	members [UnionMember]
```

Comment:
Can you talk a bit more about why you want this in here, since the information is available in `representation` already so this is a double-up? Is it simply convenience? (It is arguably much more convenient than the current layout, but the duplication bothers me a little).

Reply:
I guess this matches enums now, so it makes sense ... I've just taken for granted until now that I need to reach into `representation` to find the members and have code in various places doing this!

Reply:
I buried it in the commit message.  :)  (Read: "I will never stop hating github for hiding commit messages, until they someday, I hope, fix their horrible, horrible UI about this.")

> The rule is: the type definition must still be completely clear about
> what will happen at the type level, even if the representation field
> was stripped out or reset to a blank.  For union types: that means
> the members list must be clear, and we cannot rely on the
> representation strategy maps to contain it at some deeper level.
> 
> Yes, this means we must check for logical consistency between the
> representation strategy specification and the members list.
> But we have exactly the same kind of logical matchup work to do
> for e.g. field names in representations for structs, etc,
> so it is not unreasonable nor surprising to have that here for
> union types too.

Reply:
sgtm